src/types/UiaTextRangeBase.cpp

```diff
@@ -1009,13 +1009,21 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         textData.reserve(textDataSize);
         for (const auto& text : bufferData.text)
         {
+            if (textData.size() >= maxLength)

```

Comment:
This is an unguarded comparison of a `size_t` against a signed value. You note it below, but I don't see how this compiles under Audit or for x86.

I would VERY much rather not rely on the narrowing behavior of a signed into an unsigned. Make it explicit.

Reply:
TBH I'd just remove the loop early exit entirely. How likely is it that a client requests a buffer that's _way_ smaller than the text range? 🤔
Due to `textData.reserve(textDataSize);` the extra appends are effectively cost free too.

Reply:
I mean... Each additional row of text we copy takes just maybe ~10ns. In comparison calling `GetText` once easily takes a few 1000s of ns already and we did that even before we even get to this loop. 😔

Reply:
But _why_ should we do things inefficiently when we could just do them efficiently? I get that it's cheap to copy a bunch of text around but encouraging waste in some places encourages waste in more places. It's a hygiene thing: we leave the code better than we found it, right? You remove headers from things that don't need them, and rearrange includes, and it helps compile time _in tiny ways_. We could just keep them, but... we don't. And we get cleaner and we get better for it. 😄

Reply:
My assumption was basically that the `_start`/`_end` range isn't excessively large. Assuming so, I didn't think that copying too much is anything we have to worry about.

I suppose this would work then?
```cpp
const auto maxLengthSize = gsl::narrow_cast<size_t>(maxLength);
// ...
for (const auto& text : bufferData.text)
{
    textData.append(text, 0, maxLengthSize - textData.size());
    if (textData.size() >= maxLengthSize)
    {
        break;
    }
}
```